features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -320,19 +319,13 @@ nsapi_error_t AT_CellularContext::do_user_authentication()
         if (!get_property(PROPERTY_AT_CGAUTH)) {
             return NSAPI_ERROR_UNSUPPORTED;
         }
-        _at.cmd_start("AT+CGAUTH=");
-        _at.write_int(_cid);
-        _at.write_int(_authentication_type);
-
         const bool stored_debug_state = _at.get_debug();
         _at.set_debug(false);
 
-        _at.write_string(_uname);
-        _at.write_string(_pwd);
+        _at.at_cmd_discard("+CGAUTH", "=", "%d%d%s%s", _cid, _authentication_type, _uname, _pwd);
```

Comment:
This way AT+CGAUTH will be missing from traces and I think it is needed. handle_start could ensure that any AT command is traced if trace level allows it.

Reply:
Added mechanism to handle_start to print the command